### PR TITLE
clean & refactor udapp env list

### PR DESCRIPTION
**Pull request by yann300**
This fixes the environemebt list to make it closer to what it actually is in Live.

### CODE REVIEW

File: libs/remix-ui/run-tab/src/lib/reducers/runTab.ts [hunk 1]
```diff
@@ -252,15 +242,25 @@ export const runTabReducer = (state: RunTabState = runTabInitialState, action: A
...
-    const id = action.payload.name
-    state.providers.providerList.push({
-      content: payload.displayName,
-      dataId: payload.dataId,
-      id,
-      title: payload.title,
-      value: id
-    })
+    const payload = action.payload
+    console.log(payload)
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/reducers/runTab.ts:246]: log

File: libs/remix-ui/run-tab/src/lib/reducers/runTab.ts [hunk 2]
```diff
@@ -252,15 +242,25 @@ export const runTabReducer = (state: RunTabState = runTabInitialState, action: A
...
+    const length = state.providers.providerList.length
+    if (state.providers.providerList.length === 0) {
+      state.providers.providerList.push(payload)
+    } else {
+      let index = 0
+      for (const provider of state.providers.providerList) { 
+        if (provider.position >= payload.position) {
+          console.log('adding provider', index)
```
Comment (Aniket-Engg) [inline libs/remix-ui/run-tab/src/lib/reducers/runTab.ts:254]: here too

**Review by ryestew — CHANGES_REQUESTED** [review]
Try to switch to Ephemery.  Metamask asks to switch to Arbitrum.  
Switching from Remix VM to Injected Provider doesn't seem to work.

**Comment by Aniket-Engg** [comment]
> Try to switch to Ephemery. Metamask asks to switch to Arbitrum. Switching from Remix VM to Injected Provider doesn't seem to work.

switching to injected works fine for em

**Review by Aniket-Engg — COMMENTED** [review]
If I  try to connect ganache provider and fails, it moves to `Remix VM (Paris)`, should go to `Remix VM(cancun)` instead.

If i move to Ephemery testnet, and deny to connect on Metamask, on Remix UI, it keeps showing `Ephemery`

**Review by ryestew — APPROVED** [review]

**Pull request merged**